crates/runtime/src/lib.rs

```diff
@@ -356,22 +356,23 @@ impl_runtime_apis! {
 
     #[cfg(feature = "try-runtime")]
     impl frame_try_runtime::TryRuntime<Block> for Runtime {
-        fn on_runtime_upgrade() -> (Weight, Weight) {
+        fn on_runtime_upgrade(checks: frame_try_runtime::UpgradeCheckSelect) -> (Weight, Weight) {
```

Comment:
explain the rationale behind this new argument in comments pls

Reply:
These requirements are sourced from Substrate ver "polkadot-v0.9.43", and the code has been refactored to accommodate the following interface changes:

- The internal function signature of the `TryRuntime` trait has been modified. You can find the updated code at the following link:

    [Substrate - TryRuntime](https://github.com/paritytech/substrate/blob/polkadot-v0.9.43/frame/try-runtime/src/lib.rs)

    ```
    fn on_runtime_upgrade(checks: UpgradeCheckSelect) -> (Weight, Weight);
    ```

- The interface definition of "frame-executive" has also been updated. You can refer to the updated code at the following link:

    [Substrate - frame-executive](https://github.com/paritytech/substrate/blob/polkadot-v0.9.43/frame/executive/src/lib.rs)

    ```
    pub fn try_runtime_upgrade(
        checks: frame_try_runtime::UpgradeCheckSelect,
    ) -> Result<Weight, &'static str> {

    }
    ```

Reply:
I think he meant in the code not in a github comment

Reply:
Whoops, sorry about my bad interpretation. I've added the comment to the function description, thanks for pointing it out!